gdc_ng_models/models/cohort.py

```diff
@@ -73,15 +73,16 @@ def to_json(self):
         }
 
 
-class Cohort(Base, audit.AuditColumnsMixin):
+class Cohort(Base, audit.AuditColumnsMixin, accessed.AccessedColumnMixin):
     """A base definition for a cohort entity.
 
     Attributes:
         id: A unique identifier for the cohort.
         context_id: The ID of the associated context.
         name: A user defined name for the cohort.
         created_datetime: The date and time when the record is created.
-        updated_datetime: The date and time when the record is updated.
+        updated_datetime: The date and time when the record is last updated.
+        accessed_datetime: The date and time when the record is last accessed.
```

Comment:
I think the `updated_datetime` and `accessed_datetime` will either be the same or very close to each other. Will it make sense to re-use the `updated-datetime` instead of adding a new datetime column.
Also does the number of times a record is read affect its retention time, for example an older record more frequently accessed can be kept longer than a more recently created record that has not been accessed at all.

Reply:
The updated_datetime will only be updated if there is a mutation. The accessed_datetime will be updated anytime there is a read. So, for example, if a cohort_id is used in query by cohort, we will update the accessed datetime and the updated_datetime will remain unchanged.

Tracking the read operations will require additional updates to gdcapi/cohortapi which are planned (please view the tickets related to the story if interested). There is no way to completely handle that in the database because features like query by cohort use the ES case set associated with a cohort and there is not currently a DB read operation for that.

And no, frequency will have no bearing on retention time. We only plan on using the last accessed datetime. I don't think there has been an official decision on what that retention period is but I believe we have plans to keep it close to the expiration time of the cookie, which is 400 days.